O/ONNXRuntime/ONNXRuntimeProvidersCUDA/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "ONNXRuntimeProvidersCUDA"
+version = v"1.10.0"
+
+# Cf. https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements
+cuda_version = v"11.4"
+cudnn_version = v"8.2.4"
+
+cuda_compat = "$(cuda_version.major).$(cuda_version.minor)"
+cudnn_compat = string(cudnn_version.major)
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/microsoft/onnxruntime/releases/download/v$version/onnxruntime-linux-x64-gpu-$version.tgz", "bc880ba8a572acf79d50dcd35ba6dd8e5fb708d03883959ef60efbc15f5cdcb6"; unpack_target="onnxruntime-x86_64-linux-gnu"),
+    ArchiveSource("https://github.com/microsoft/onnxruntime/releases/download/v$version/onnxruntime-win-x64-gpu-$version.zip", "0da11b8d953fad4ec75f87bb894f72dea511a3940cff2f4dad37451586d1ebbc"; unpack_target="onnxruntime-x86_64-w64-mingw32")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+
+if [[ $target == *-w64-mingw32* ]]; then
+    chmod 755 onnxruntime-$target/onnxruntime-*/lib/*
+fi
+mkdir -p $includedir $libdir
+cp -av onnxruntime-$target/onnxruntime-*/include/* $includedir
+cp -av onnxruntime-$target/onnxruntime-*/lib/*onnxruntime_providers_cuda* $libdir
+install_license onnxruntime-$target/onnxruntime-*/LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "Linux"; cuda = cuda_compat),
+    Platform("x86_64", "Windows"; cuda = cuda_compat)
```

Comment:
@maleadt Should a package dependent on CUDA/CUDNN assign cuda_compat like so (see also CUDNN dependency)?

Reply:
This part of the dependency chain isn't functional yet, and requires (at the least) https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/1128. You'll have to use CUDA_full_jll for now, and do your own runtime dependency management with CUDA.jl.

Reply:
Not sure I quite understood this. It seems to be that adding the `cuda` platform augmentation makes sure that CUDA_loader (added via CUDNN) will select the `cuda_compat` version?

Reply:
Yes, we could add a platform augmentation to CUDA_loader_jll to make it load the correct CUDA toolkit for a given driver version. However, CUDNN_jll then needs to load an appropriate artifact based on the loaded toolkit, and not the available driver (a driver supporting CUDA 11.3 can work with CUDA toolkit 11.6, because of semver, and you then need CUDNN for CUDA 11.6). That hasn't been fully figured out yet, so we don't have any platform augmentation in place yet (also because it's a brand new feature).

Reply:
@maleadt Not sure I quite understood this (still), but I have now added builds/variants for each CUDA major-minor release. Does this look right?

... or should `CUDA_Runtime_jll` be used?

Reply:
What I meant is that all this just doesn't Just Work yet, somebody needs to put in some time to figure out a good way to use platform augmentation for CUDA dependencies (and I haven't had the time yet, although CUDA_Runtime_jll is a step in that direction). For now, you'll need to load your artifact manually, ensuring the necessary/correct dependencies have been loaded first. This is what CUDA.jl currently does.

Reply:
It does sound right to me, but OTOH I am not an expert on building CUDA and have never used advanced Artifact platform features.

Reply:
Sounds about right.

> `ORT.load_inference` with `execution_provider=:cuda` must then make sure to load the proper `ONNXRuntime_jll` artifact - by selecting the proper artifact for e.g. CUDA 10 and CUDA 11, by doing something like https://pkgdocs.julialang.org/v1/artifacts/#Extending-Platform-Selection

Make sure you have the platform augmentation rely on `CUDA.runtime_version()` here, as that may be different from the maximally supported version by the driver as reported by `CUDA.version()`.

Reply:
Fixed TODOs with a81fa6b and f3dbc70.